services/core/shared/core/compat/sdk_v5/transactions.js

```diff
@@ -72,42 +78,58 @@ const normalizeTransaction = tx => {
 	return tx;
 };
 
-const getTransactions = async params => {
-	const transactionsDB = await knex('transactions');
-	const transactions = {
-		data: [],
-		meta: {},
-	};
+const validateParams = async params => {
 	if (params.fromTimestamp || params.toTimestamp) {
 		params.propBetween = {
 			property: 'timestamp',
 			from: Number(params.fromTimestamp) || 0,
 			to: Number(params.toTimestamp) || Math.floor(Date.now() / 1000),
 		};
 	}
+	if (params.sort && params.sort.includes('nonce')) {
+		if (!params.senderId) return new Error('Nonce based sorting is only possible along with senderId');
+	}
+	if (params.senderId) params.senderPublicKey = await getPublicKeyByAddress(params.senderId);
+	delete params.senderId;
 	if (params.moduleAssetName) params.moduleAssetId = resolveModuleAsset(params.moduleAssetName);
 	delete params.moduleAssetName;
+	return params;
+};
+
+const getTransactions = async params => {
+	const transactionsDB = await knex('transactions');
+	const transactions = {
+		data: [],
+		meta: {},
+	};
+
+	params = await validateParams(params);
 
-	// TODO: Add check to ensure nonce based sorting always requires senderId,
-	// Update once account index is implemented.
 	const resultSet = await transactionsDB.find(params);
 	if (resultSet.length) params.ids = resultSet.map(row => row.id);
-	const response = await coreApi.getTransactions(params);
-	if (response.data) transactions.data = response.data.map(tx => normalizeTransaction(tx));
-	if (response.meta) transactions.meta = response.meta;
+	if (params.ids || params.id) {
+		const response = await coreApi.getTransactions(params);
+		if (response.data) transactions.data = response.data.map(tx => normalizeTransaction(tx));
+		if (response.meta) transactions.meta = response.meta;
 
-	transactions.data = await BluebirdPromise.map(
-		transactions.data,
-		async transaction => {
-			const [indexedTxInfo] = resultSet.filter(tx => tx.id === transaction.id);
-			transaction.unixTimestamp = indexedTxInfo.timestamp;
-			transaction.height = indexedTxInfo.height;
-			transaction.blockId = indexedTxInfo.blockId;
-
-			return transaction;
-		},
-		{ concurrency: transactions.data.length },
-	);
+		transactions.data = await BluebirdPromise.map(
+			transactions.data,
+			async transaction => {
+				const [indexedTxInfo] = resultSet.filter(tx => tx.id === transaction.id);
+				transaction.unixTimestamp = indexedTxInfo.timestamp;
+				transaction.height = indexedTxInfo.height;
+				transaction.blockId = indexedTxInfo.blockId;
+				const [{
+					address,
+					username,
+				}] = await getIndexedAccountByPublicKey(transaction.senderPublicKey);
+				transaction.senderId = address;
+				transaction.username = username || undefined;
```

Comment:
Consider adding null checks.
```suggestion
				const [response] = await getIndexedAccountByPublicKey(transaction.senderPublicKey);
				transaction.senderId = response && response.address ? response.address : undefined;
				transaction.username = response && response.username ? response.username : undefined;
```